FWCore/ParameterSet/python/SequenceTypes.py

```diff
@@ -1626,51 +1670,136 @@ def remove(self, something):
     def resolve(self, processDict,keepIfCannotResolve=False):
         temp = OrderedSet()
         for i in self._collection:
-            if isinstance(i, Task) or isinstance(i, TaskPlaceholder):
+            if self._mustResolve(i):
                 temp.add(i.resolve(processDict,keepIfCannotResolve))
             else:
                 temp.add(i)
         self._collection = temp
         return self
-
-class TaskPlaceholder(object):
+        
+class _TaskBasePlaceholder(object):
     def __init__(self, name):
         self._name = name
     def _isTaskComponent(self):
-        return True
+        return False
     def isLeaf(self):
         return False
     def visit(self,visitor):
         pass
     def __str__(self):
         return self._name
     def insertInto(self, parameterSet, myname):
-        raise RuntimeError("The TaskPlaceholder "+self._name
-                           +" was never overridden")
+        raise RuntimeError("The {} {} was never overridden".format(self._typeName(), self._name))
     def resolve(self, processDict,keepIfCannotResolve=False):
         if not self._name in processDict:
             if keepIfCannotResolve:
                 return self
-            raise RuntimeError("The TaskPlaceholder "+self._name+ " cannot be resolved.\n Known keys are:"+str(processDict.keys()))
+            raise RuntimeError("The {} {} cannot be resolved.\n Known keys are: {}".format(self._typeName(), self._name,str(processDict.keys())))
         o = processDict[self._name]
-        if not o._isTaskComponent():
-            raise RuntimeError("The TaskPlaceholder "+self._name+ " refers to an object type which is not allowed to be on a task: "+str(type(o)))
-        if isinstance(o, Task):
+        if not self._allowedInTask(o):
+            raise RuntimeError("The {} {} refers to an object type which is not allowed to be on a task: {}".format(self._typeName(), self._name, str(type(o))))
+        if isinstance(o, self._taskClass()):
             return o.resolve(processDict)
         return o
     def copy(self):
-        returnValue =TaskPlaceholder.__new__(type(self))
-        returnValue.__init__(self._name)
-        return returnValue
+        return self._makeInstance(self._name)
     def dumpSequencePython(self, options=PrintOptions()):
-        return 'cms.TaskPlaceholder("%s")'%self._name
+        return 'cms.{}("{}")'.format(self._typeName(), self._name)
     def dumpPython(self, options=PrintOptions()):
-        result = 'cms.TaskPlaceholder(\"'
+        result = 'cms.{}(\"'.format(self._typeName())
         if options.isCfg:
            result += 'process.'
         result += self._name+'\")\n'
         return result
 
+class Task(_TaskBase) :
+    """Holds EDProducers, EDFilters, ESProducers, ESSources, Services, and Tasks.
+    A Task can be associated with Sequences, Paths, EndPaths and the Schedule.
```

Comment:
Right,  but it can be associated with `ConditionalTask`. Or would some other wording than "associated" be more precise for the relationship between `Task` and `ConditionalTask`?